contracts/core/BorrowableMarginPool.sol

```diff
@@ -0,0 +1,288 @@
+     * @param _optionsVault address of the options vault
+     * @param _whitelisted bool of whether it is whitelisted or blacklisted
+     */
+    function setOptionsVaultWhitelistedStatus(address _optionsVault, bool _whitelisted) external onlyOwner {
+        require(_optionsVault != address(0), "!_optionsVault");
+
+        whitelistedOptionsVault[_optionsVault] = _whitelisted;
+        emit SetOptionsVaultWhitelist(_optionsVault, _whitelisted);
+    }
+
+    /**
+     * @notice Set oToken buyer whitelist status
+     * @param _oTokenBuyer address of the oToken buyer
+     * @param _whitelisted bool of whether it is whitelisted or blacklisted
+     */
+    function setOTokenBuyerWhitelistedStatus(address _oTokenBuyer, bool _whitelisted) external onlyOwner {
+        require(_oTokenBuyer != address(0), "!_buyer");
+
+        whitelistedOTokenBuyer[_oTokenBuyer] = _whitelisted;
+        emit SetOTokenBuyerWhitelist(_oTokenBuyer, _whitelisted);
+    }
+
+    /**
+     * @notice Set Borrow percent of collateral asset
+     * @param _collateral address of collateral asset
+     * @param _borrowPCT borrow PCT
+     */
+    function setBorrowPCT(address _collateral, uint256 _borrowPCT) external onlyOwner {
+        borrowPCT[_collateral] = _borrowPCT;
+        emit SetBorrowPCT(_collateral, _borrowPCT);
+    }
+
+    /**
+     * @notice Lends out asset to market maker
+     * @param _oToken address of the oToken laying claim to the collateral assets
+     * @param _amount amount of the asset to lend to borrower
+     */
+    function borrow(address _oToken, uint256 _amount) public onlyWhitelistedBorrower {
+        require(
+            WhitelistInterface(AddressBookInterface(addressBook).getWhitelist()).isWhitelistedOtoken(_oToken),
+            "MarginPool: oToken is not whitelisted"
+        );
+
+        require(_amount > 0, "MarginPool: Cannot borrow 0 of underlying");
+
+        OtokenInterface oToken = OtokenInterface(_oToken);
+
+        address collateralAsset = oToken.collateralAsset();
+        uint256 outstandingAssetBorrow = borrowed[msg.sender][collateralAsset];
+
+        require(!oToken.isPut(), "MarginPool: oToken is not a call option");
+
+        // Make sure borrow does not attempt to borrow with an expired oToken
+        require(
+            oToken.expiryTimestamp() > block.timestamp,
+            "MarginPool: Cannot borrow collateral asset of expired oToken"
+        );
+
+        uint256 collateralAssetDecimals = ERC20Interface(collateralAsset).decimals();
+
+        uint256 oTokenBalance = ERC20Interface(_oToken).balanceOf(msg.sender);
+
+        // Each oToken represents 1:1 of collateral token
+        // So a market maker can borrow at most our oToken balance in the collateral asset
+        uint256 collateralAllocatedToMM = collateralAssetDecimals >= 8
+            ? oTokenBalance.mul(10**(collateralAssetDecimals.sub(8)))
+            : oTokenBalance.div(10**(uint256(8).sub(collateralAssetDecimals)));
+
```

Comment:
Can u explain more the >= 8 decimals ? maybe also add some inline comments for it

Reply:
oTokens are 8 decimals. 

If the collateral token X for the oToken has decimals >= 8, we need to scale the oToken balance up to match "the equivalent borrowable in the asset" since each oToken is one underlying. Conversely if collateral token has decimals <8, we need to scale the oToken balance down to match how much in underlying is actually borrowable